src/core/libmaven/isotopeDetection.cpp

```diff
@@ -63,207 +63,182 @@ void IsotopeDetection::pullIsotopes(PeakGroup* parentgroup)
-            return true;
-    }
+void IsotopeDetection::filterIsotopePeaks(const Peak* parentPeak,
+                                          vector<Peak>& isotopePeaks,
+                                          mzSample* sample)
+{
+    if (parentPeak == nullptr)
+        return;
 
-    //TODO: this is really an abuse of the maxIsotopeScanDiff parameter
-    //I can easily imagine you might set maxIsotopeScanDiff to something much less than the peak width
-    //here w should really be determined by the minRt and maxRt for the parent and child peaks
-    if (parentGroup)
-    {
-        float isotopeMass = x.mass;
-        Compound* compound = parentGroup->getCompound();
-        float parentMass = MassCalculator::computeMass(compound->formula(),
-                                                       _mavenParameters->getCharge(compound));
-
-        Peak* parentPeak = parentGroup->getPeak(sample);
-        float rtmin = parentGroup->minRt;
-        float rtmax = parentGroup->maxRt;
-        if (parentPeak)
-        {
-            rtmin = parentPeak->rtmin;
-            rtmax = parentPeak->rtmax;
-        }
-        float w = _mavenParameters->maxIsotopeScanDiff
-            * _mavenParameters->avgScanTime;
-        double c = sample->correlation(
-                isotopeMass, parentMass,
-                _mavenParameters->compoundMassCutoffWindow, rtmin - w,
-                rtmax + w, _mavenParameters->eicType,
-            _mavenParameters->filterline);  // find correlation for isotopes
-        if (c < _mavenParameters->minIsotopicCorrelation)
+    auto badCorrelation = [&] (const Peak& peak) {
+        auto corr = sample->correlation(parentPeak->mzmin,
+                                        parentPeak->mzmax,
+                                        peak.mzmin,
+                                        peak.mzmax,
+                                        parentPeak->rtmin,
+                                        parentPeak->rtmax,
+                                        1, // ms level
+                                        _mavenParameters->eicType,
+                                        _mavenParameters->filterline);
+        if (corr < _mavenParameters->minIsotopicCorrelation)
             return true;
-    }
-    return false;
+        return false;
+    };
+    isotopePeaks.erase(remove_if(begin(isotopePeaks),
+                                 end(isotopePeaks),
+                                 badCorrelation),
+                       end(isotopePeaks));
 }
 
 std::pair<float, float> IsotopeDetection::getIntensity(Scan* scan, float mzmin, float mzmax)
 {
-    float highestIntensity = 0;
-    float rt = 0;
+    float highestIntensity = 0.0f;
+    float rt = 0.0f;
     mzSample* sample = scan->getSample();
 
-    for (int i = scan->scannum - 2; i < scan->scannum + 2; i++) {
+    for (int i = scan->scannum - _mavenParameters->maxIsotopeScanDiff;
+         i < scan->scannum + _mavenParameters->maxIsotopeScanDiff;
```

Comment:
`<=` ? I, as a user would expect +-maxIsotopeScanDiff around the peak, but it's not a big deal. Your call.

Reply:
No, `<=` makes more sense.